cmd/server/main.go

```diff
@@ -139,6 +141,15 @@ func buildCLI() *cli.App {
 					return cli.Exit(fmt.Sprintf("Unable to load configuration: %v.", err), 1)
 				}
 
+				if !c.IsSet("service") && !c.IsSet("services") {
+					if _, ok := cfg.Services[string(primitives.InternalFrontendService)]; !ok {
+						// --services is using default value (all services) but internal-frontend
```

Comment:
Where do we set the default values for services? Why not change the default value to be explicitly the 4 roles?

Reply:
The default value is just https://github.com/temporalio/temporal/blob/master/temporal/server.go#L49-L54 , i.e. the same list that is used for validation.

So, that idea would be to add a new default value with just the four. There'd be a chance of adding a new role and forgetting to add it to the default list, but I guess if they're right next to each other that seems minimal

Reply:
(When I was working on this I had the intention of migrating everyone to use internal-frontend by default over a couple releases. It's more consistent and simpler. In that world, the original contents of this PR would be just for compatibility and could be deleted after a while, and keeping the default be "all" makes sense. But I guess we don't have consensus to go that way and it might not be worth it.)